CMSgov/mint-app
# [EASI-4504] Improve local user accounts

# EASI-4504

## Description

Before this PR, all test user data followed the pattern user_name + "Doe". This was not the best experience for demos etc. This PR aimed to expose other test data so it would be more meaningful for demos and local development.

- Replace the mock user data call by instead wiring up the local Okta API client.
- getMockUserData replaced with a dictionary variable.
- Additional mock user data added to include values seen in FIGMA as well as other additional test names

## How to test this change

1.  bring up the app
2. log in with local login as any user that has test data in `pkg/local/okta_api.go`.
3. Verify that  a User account is made that matches the data provided in the local files
4. Log out
5. Log in with an EUAID that isn't in the list, verify that you can still log in with the old UserName + "Doe" paradigm.

Collaborator
1. Search for users in the list
2. Add as a collaborator

Discussions
1. Tag users from the local dev data in a discussion. 
2. Verify it looks correct

Tests
1. Verify that all the tests continue to function as expected. 

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
